src/main/java/org/jenkinsci/plugins/mesos/JenkinsScheduler.java

```diff
@@ -407,6 +408,28 @@ private void createMesosTask(Offer offer, Request request) {
     LOGGER.info("Launching task " + taskId.getValue() + " with URI " +
                 joinPaths(jenkinsMaster, SLAVE_JAR_URI_SUFFIX));
 
+    for (final Computer computer : Jenkins.getInstance().getComputers()) {
```

Comment:
We are, you can see that the slave name is the name of the mesos task. It just so happens that you can be offered a task after it has been started or deleted which leads to orphan tasks and containers

On June 21, 2015 3:10:49 PM CDT, Vinod Kone <notifications@github.com> wrote:
>> @@ -407,6 +408,28 @@ private void createMesosTask(Offer offer,
>Request request) {
>>      LOGGER.info("Launching task " + taskId.getValue() + " with URI "
>+
>>                  joinPaths(jenkinsMaster, SLAVE_JAR_URI_SUFFIX));
>>  
>> +    for (final Computer computer :
>Jenkins.getInstance().getComputers()) {
>
>I'm not sure I follow the logic here. Why are we creating a mesos task
>with the same name as an existing jenkins slave in the first place? 
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/jenkinsci/mesos-plugin/pull/121/files#r32896961

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.